lib/src/builders/file_builder.dart

```diff
@@ -36,33 +40,41 @@ class ExportBuilder extends _AbstractCodeBuilder<ExportDirective> {
 /// Builds files of Dart source code.
 ///
 /// See [LibraryBuilder] and [PartBuilder] for concrete implementations.
-abstract class FileBuilder extends _AbstractCodeBuilder<CompilationUnit> {
-  FileBuilder._(CompilationUnit astNode) : super._(astNode);
+abstract class FileBuilder implements CodeBuilder<CompilationUnit> {
+  final List<CodeBuilder<Declaration>> _declarations =
+      <CodeBuilder<Declaration>>[];
+
+  FileBuilder._();
 
   /// Adds [declaration]'s resulting AST to the source.
   void addDeclaration(CodeBuilder<Declaration> declaration) {
-    _astNode.declarations.add(declaration.toAst());
+    _declarations.add(declaration);
   }
+
+  @override
+  @mustCallSuper
+  CompilationUnit toAst([Scope scope = const Scope.identity()]) =>
+      _emptyCompilationUnit()
+        ..declarations
+            .addAll(_declarations.map/*<Declaration>*/((d) => d.toAst(scope)));
 }
 
 /// An `import` directive in a [FileBuilder].
-class ImportBuilder extends _AbstractCodeBuilder<ImportDirective> {
-  static Token _as = new KeywordToken(Keyword.AS, 0);
+class ImportBuilder implements CodeBuilder<ImportDirective> {
+  final String _uri;
+  final String _prefix;
 
   /// Create a new `import` directive importing [uri].
   ///
   /// Optionally prefix [as].
-  factory ImportBuilder(String uri, {String as}) {
-    var astNode = _createImportDirective()..uri = _stringLiteral("'$uri'");
-    if (as != null) {
-      astNode
-        ..asKeyword = _as
-        ..prefix = _stringIdentifier(as);
-    }
-    return new ImportBuilder._(astNode);
-  }
+  const factory ImportBuilder(String uri, {String as}) = ImportBuilder._;
```

Comment:
`as` is a Dart keyword. How about `withPrefix` or just `prefix`? That's what the language spec calls them I think.